src/entity.ts

```diff
@@ -607,6 +591,38 @@ export namespace entity {
       return valueProto;
     }
 
+    if (typeof value === 'number') {
+      const integerOutOfBoundsWarning =
+        "'IntegerOutOfBoundsWarning: the value for '" +
+        property +
+        "' property is outside of bounds of a JavaScript Number.\n" +
+        "Use 'Datastore.int(<integer_value_as_string>)' to preserve accuracy during the upload.";
+
+      const typeCastWarning =
+        "TypeCastWarning: the value for '" +
+        property +
+        "' property is a JavaScript Number.\n" +
+        "Use 'Datastore.int(<integer_value_as_string>)' or " +
+        "'Datastore.double(<double_value_as_string>)' to preserve consistent " +
+        'Datastore types during the upload.';
+
+      if (Number.isInteger(value)) {
+        if (!Number.isSafeInteger(value)) {
+          process.emitWarning(integerOutOfBoundsWarning);
+        } else {
```

Comment:
Do we need the else here ? 
I don't think the branches are exclusive.

Also maybe move `process.emitWarning(typeCastWarning);` right after l. 594 as soon as we know that the type is a number ?

Reply:
They are exclusive. They both recommend the same outcome, but I think having two separate warnings is a bit confusing. This one warning is it could be dangerous, the other is destructive. I had what you recommended earlier but it is very very noisy :)

Reply:
Note: added 'debouncing' you will only get a single typecast warning for the entity. that way it doesn't flood the warning feed.

Reply:
It might be reasonable to augment the out of bounds warning to mention the type issue as well, since that'll still be an issue if they fix their number size issue.

Reply:
Since we're debouncing the typescast warning any ways, I'd consider just dropping the if and printing both.

Reply:
I think printing both could get noisy as Chris said. I ended up going with @feywind's suggestion here.